lib/commands/source.js

```diff
@@ -37,6 +42,70 @@ helpers.mobileGetSource = async function (opts = {}) {
   return await this.proxyCommand(`/source?format=${encodeURIComponent(opts.format)}`, 'GET');
 };
 
+/* Will get JSON of the form:
+ *   { isEnabled: '1',
+ *     isVisible: '1',
+ *     frame: '{{0, 0}, {375, 667}}',
+ *     children:
+ *      [ { isEnabled: '1',
+ *          isVisible: '1',
+ *          frame: '{{0, 0}, {375, 667}}',
+ *          children: [],
+ *          rect: { x: 0, y: 0, width: 375, height: 667 },
+ *          value: null,
+ *          label: null,
+ *          type: 'Other',
+ *          name: null,
+ *          rawIdentifier: null },
+ *     rect: { origin: { x: 0, y: 0 }, size: { width: 375, height: 667 } },
+ *     value: null,
+ *     label: 'UICatalog',
+ *     type: 'Application',
+ *     name: 'UICatalog',
+ *     rawIdentifier: null }
+ */
+function getTreeForXML (srcTree) {
+  function getTree (element, elementIndex, parentPath) {
+    let curPath = `${parentPath}/${elementIndex}`;
+    let rect = element.rect || {};
+    let subtree = {
+      '@': {
+        name: element.name || '',
```

Comment:
the corresponding attribute is not shown in the original xml if the value of it is `null`

Reply:
the same is true for other nullable attributes below